### PR TITLE
awful.client.history: allow to disable history tracking

### PR DESCRIPTION
**Pull request by blueyed**
Based on the code from Uli Schlachter at
https://github.com/awesomeWM/awesome/issues/572#issuecomment-160080378.

Fixes https://github.com/awesomeWM/awesome/issues/572.

### CODE REVIEW

**Comment by blueyed** [comment]
Does the same make sense for autofocus, too?

File: lib/awful/client.lua [hunk 1]
```diff
@@ -1142,9 +1140,50 @@ capi.client.connect_signal("manage", function (c)
...
+        end
+        return disabled_count == 0
+    end
+
+    --- Is history tracking enabled?
+    -- @treturn[0] true True is history tracking is enabled.
+    -- @treturn[1] int The number of times that tracking has been disabled.
+    -- @function awful.client.focus.history.is_enabled
+    function client.focus.history.is_enabled()
+        return disabled_count == 0 or disabled_count
```
Comment (psychon) [inline lib/awful/client.lua]: Did you mean `,` instead of `or`?

Comment (blueyed) [inline lib/awful/client.lua]: No.. it should have been either `true` (`disabled_count` == 0), or `disabled_count`.
What you mean is returning two values `bool` and the number, right?
Makes more sense and seems to be less magical that way - amended.

Comment (psychon) [inline lib/awful/client.lua]: Apparently I still have problems understanding `@return[42]`.

Yup, you are right and yup, I am right. :-)

**Pull request merged**